### ISSUE TITLE
strange errors with truenas scale

### ISSUE THREAD

**Issue by djbobyd** [democratic-csi/democratic-csi#195]
Hi Travis,
After upgrading my storage server to TrueNAS Scale 22.02.1 I am experiencing some strange errors with the latest CSI driver. 
When a volume is created or deleted the below errors appear in the logs and the csi-controller restarts. At the same time many of the volumes for my other statefulsets become read-only. 
I checked the logs of the iscsi service on the truenas server but there are no errors there. Same is for the logs of the iscsi client service on the  kubernetes nodes. 

Do you have any ideas what else can i check to see what the problem might be?

```
I0525 18:54:11.948231       1 connection.go:187] GRPC error: rpc error: code = Unavailable desc = unexpected HTTP status code received from server: 502 (Bad Gateway); malformed header: missing HTTP content-type
E0525 18:54:11.948265       1 controller.go:1481] delete "pvc-0f6b10ec-2d49-4215-a51a-b94fd2f49efa": volume deletion failed: rpc error: code = Unavailable desc = unexpected HTTP status code received from server: 502 (Bad Gateway); malformed header: missing HTTP content-type
W0525 18:54:11.948298       1 controller.go:989] Retrying syncing volume "pvc-0f6b10ec-2d49-4215-a51a-b94fd2f49efa", failure 4
E0525 18:54:11.948319       1 controller.go:1007] error syncing volume "pvc-0f6b10ec-2d49-4215-a51a-b94fd2f49efa": rpc error: code = Unavailable desc = unexpected HTTP status code received from server: 502 (Bad Gateway); malformed header: missing HTTP content-type
I0525 18:54:11.948352       1 event.go:285] Event(v1.ObjectReference{Kind:"PersistentVolume", Namespace:"", Name:"pvc-0f6b10ec-2d49-4215-a51a-b94fd2f49efa", UID:"a9b0cb1d-b997-4c6c-babc-0bf04cef3c55", APIVersion:"v1", ResourceVersion:"506885956", FieldPath:""}): type: 'Warning' reason: 'VolumeFailedDelete' rpc error: code = Unavailable desc = unexpected HTTP status code received from server: 502 (Bad Gateway); malformed header: missing HTTP content-type
```

```
I0525 05:01:09.738873       1 feature_gate.go:245] feature gates: &{map[]}
I0525 05:01:09.738953       1 csi-provisioner.go:139] Version: v3.1.0
I0525 05:01:09.738960       1 csi-provisioner.go:162] Building kube configs for running in cluster...
I0525 05:01:09.745901       1 connection.go:154] Connecting to unix:///csi-data/csi.sock
I0525 05:01:11.762817       1 common.go:111] Probing CSI driver for readiness
I0525 05:01:11.763279       1 connection.go:183] GRPC call: /csi.v1.Identity/Probe
I0525 05:01:11.763297       1 connection.go:184] GRPC request: {}
I0525 05:01:31.772525       1 connection.go:186] GRPC response: {}
I0525 05:01:31.772764       1 connection.go:187] GRPC error: rpc error: code = Internal desc = Error: Timed out while waiting for handshake Error: Timed out while waiting for handshake
    at Timeout._onTimeout (/home/csi/app/node_modules/ssh2/lib/client.js:1014:23)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
E0525 05:01:31.772814       1 csi-provisioner.go:197] CSI driver probe failed: rpc error: code = Internal desc = Error: Timed out while waiting for handshake Error: Timed out while waiting for handshake
    at Timeout._onTimeout (/home/csi/app/node_modules/ssh2/lib/client.js:1014:23)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
```

Thanks!

**Comment by travisghansen** [comment]
Can you send over the logs from the `csi-driver` container? What version of the driver are you running and what version of the chart did you use to install with?

**Comment by djbobyd** [comment]
Sure, here is the log
[logs-from-csi-driver-in-zfs-iscsi-democratic-csi-controller-cb6cc7c68-crz8z.log](https://github.com/democratic-csi/democratic-csi/files/8774340/logs-from-csi-driver-in-zfs-iscsi-democratic-csi-controller-cb6cc7c68-crz8z.log)

and this is the version of the chart: 0.12.0
driver should be: 1.5.0 as I haven't modified the default in the chart
 I first thought it might be from the api driver but then i switched back to using ssh and the errors continued.

**Comment by travisghansen** [comment]
Maybe enable debug logging...it seems the driver is having issues communicating with the truenas api.

```
controller:
  driver:
    logLevel: debug

node:
  driver:
    logLevel: debug
```

**Comment by travisghansen** [comment]
Can you create a new target manually via the UI? I'm wondering if the TN database is in a weird state.

If you restart the NAS the mounted PVs will switch to `ro` as that is what `iscsid` will do if the connection is dropped and comes back up. In those cases the best thing to do is just delete any relevant pods and let them recreate.

**Comment by djbobyd** [comment]
I don't see any NAS restarts but I do see something strange when creating new target. everything gets disconnected and connected again. I.ve attached the relevant logs.
[nas.log](https://github.com/democratic-csi/democratic-csi/files/8784019/nas.log)

there is unfortunately no info why is this happening. everything seems to be working normally, at least from what i can see in the logs.

**Comment by djbobyd** [comment]
Ok, I think I fixed it. It might be a problem with the upgrade procedure of TrueNAS SCALE.
Basically the problem was that some of the extents created in the past have been using 
```
extentRpm: "7200"
```
configuration of the iscsi driver
newer extents were created with
```
extentRpm: "SSD"
```
This was working fine with TrueNAS CORE. Thus I never cared about this config. However after seeing the disconnects, attached in the previous post, I started looking into SCST daemon and played around with the console utility.
You can check the essentials in this file [findings.txt](https://github.com/democratic-csi/democratic-csi/files/8789515/findings.txt)

The bottom line is that it seems that 7200 configuration is not working as expected. I changed all wrong extents and now there are no issues creating or destroying PVs. 
Hope this is helpful for you or someone else having simillar issue.

**Comment by travisghansen** [comment]
Awesome! Yeah we need to report this upstream for sure.

**Comment by travisghansen** [comment]
https://jira.ixsystems.com/browse/NAS-116546

**Comment by djbobyd** [comment]
Thanks for reporting that. I was just wondering how to do it myself. 
Since there is a fix already submitted and you know about the issue, should we already close this issue?
Thnaks again for the support!

**Comment by travisghansen** [comment]
I’ll leave it open until the other is confirmed fixed etc

**Comment by travisghansen** [comment]
OK, it was merged (to the 22.12 SCALE release). I'll close this but if problems show up we can re-open later down the road.
